GPL/Events/Process/Probe.bpf.c

```diff
@@ -226,6 +257,195 @@ int tracepoint_syscalls_sys_exit_setsid(struct trace_event_raw_sys_exit *args)
     return 0;
 }
 
+SEC("tp_btf/module_load")
+int BPF_PROG(module_load, struct module *mod)
+{
+    if (ebpf_events_is_trusted_pid())
+        goto out;
+
+    struct ebpf_process_load_module_event *event = get_event_buffer();
+    if (!event)
+        goto out;
+
+    event->hdr.type = EBPF_EVENT_PROCESS_LOAD_MODULE;
+    event->hdr.ts   = bpf_ktime_get_ns();
+
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+
+    if (is_kernel_thread(task))
+        goto out;
+
+    ebpf_pid_info__fill(&event->pids, task);
+
+    pid_t ppid      = BPF_CORE_READ(task, group_leader, real_parent, tgid);
+    pid_t curr_tgid = BPF_CORE_READ(task, tgid);
+
+    // ignore if process is child of init/systemd/whatever
+    if ((curr_tgid == 1) || (curr_tgid == 2) || (ppid == 1) || (ppid == 2))
+        goto out;
+
+    // Variable length fields
+    ebpf_vl_fields__init(&event->vl_fields);
+    struct ebpf_varlen_field *field;
+    long size;
+
+// from include/linux/moduleparam.h
+#define MAX_PARAM_PREFIX_LEN (64 - sizeof(unsigned long))
+
+    // mod name
+    field = ebpf_vl_field__add(&event->vl_fields, EBPF_VL_FIELD_FILENAME);
+    size  = read_kernel_str_or_empty_str(field->data, MAX_PARAM_PREFIX_LEN, mod->name);
+    ebpf_vl_field__set_size(&event->vl_fields, field, size);
+
+    // mod version
+    field = ebpf_vl_field__add(&event->vl_fields, EBPF_VL_FIELD_MOD_VERSION);
+    size  = read_kernel_str_or_empty_str(field->data, PATH_MAX, mod->version);
+    ebpf_vl_field__set_size(&event->vl_fields, field, size);
+
+    // mod srcversion
+    field = ebpf_vl_field__add(&event->vl_fields, EBPF_VL_FIELD_MOD_SRCVERSION);
+    size  = read_kernel_str_or_empty_str(field->data, PATH_MAX, mod->srcversion);
+    ebpf_vl_field__set_size(&event->vl_fields, field, size);
+
+    bpf_ringbuf_output(&ringbuf, event, EVENT_SIZE(event), 0);
+
+out:
+    return 0;
+}
+
+SEC("kprobe/ptrace_attach")
+int BPF_KPROBE(kprobe__ptrace_attach,
+               struct task_struct *child,
+               long request,
+               unsigned long addr,
+               unsigned long flags)
+{
+    if (ebpf_events_is_trusted_pid())
+        goto out;
+
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    pid_t curr_tgid          = BPF_CORE_READ(task, tgid);
+    pid_t child_ppid         = BPF_CORE_READ(child, group_leader, real_parent, tgid);
+    pid_t child_tgid         = BPF_CORE_READ(child, tgid);
+
+    if (is_kernel_thread(task))
+        goto out;
+
```

Comment:
```suggestion
    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
    if (is_kernel_thread(task))
        goto out;

    pid_t curr_tgid          = BPF_CORE_READ(task, tgid);
    pid_t child_ppid         = BPF_CORE_READ(child, group_leader, real_parent, tgid);
    pid_t child_tgid         = BPF_CORE_READ(child, tgid);
```
Don't bother reading things if we're about to bail

Reply:
done